hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusCreateTopicHandler.java

```diff
@@ -174,6 +173,63 @@ public void handle(@NonNull final HandleContext handleContext) {
         }
     }
 
+    private void validateSemantics(
+            ConsensusCreateTopicTransactionBody op, HandleContext handleContext, Topic.Builder builder) {
+
+        final var configuration = handleContext.configuration();
+        final var topicConfig = configuration.getConfigData(TopicsConfig.class);
+        final var topicStore = handleContext.storeFactory().readableStore(ReadableTopicStore.class);
+        final var accountStore = handleContext.storeFactory().readableStore(ReadableAccountStore.class);
+        final var tokenStore = handleContext.storeFactory().readableStore(ReadableTokenStore.class);
+        final var tokenRelStore = handleContext.storeFactory().readableStore(ReadableTokenRelationStore.class);
+
+        // Validate admin and submit keys and set them. Empty key list is allowed and is used for immutable entities
+        if (op.hasAdminKey() && !isImmutableKey(op.adminKey())) {
+            handleContext.attributeValidator().validateKey(op.adminKey());
+            builder.adminKey(op.adminKey());
+        }
+
+        // submitKey() is not checked in preCheck()
+        if (op.hasSubmitKey()) {
+            handleContext.attributeValidator().validateKey(op.submitKey());
+            builder.submitKey(op.submitKey());
+        }
+
+        // validate hasFeeScheduleKey()
+        if (op.hasFeeScheduleKey()) {
+            handleContext.attributeValidator().validateKey(op.feeScheduleKey(), INVALID_CUSTOM_FEE_SCHEDULE_KEY);
+            builder.feeScheduleKey(op.feeScheduleKey());
+        }
+
+        // validate size of the list and the keys
+        if (!op.feeExemptKeyList().isEmpty()) {
```

Comment:
We don't need to check `if (!op.feeExemptKeyList().isEmpty())` right ? All the below conditions can be outside the if

Reply:
Fixed.